romana/kubernetes/listener.go

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// Listens to Kubernetes for policy updates
+package kubernetes
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/romana/core/common"
+	"github.com/romana/core/tenant"
+	"io"
+	"log"
+	"strings"
+	//	"encoding/json"
+	//	"time"
+	//	"net/http"
+	//	"runtime"
+)
+
+// readChunk reads the next chunk from the provided reader.
```

Comment:
And what was wrong with `json.Decoder` ?
It handles chunked input automatically and I think implementing chunked read won't be much appreciated by Go people.
https://github.com/romana/contrib/blob/dev/rsearch/rsearch/resources.go#L89